core/src/main/java/de/bwaldvogel/mongo/oplog/CollectionBackedOplog.java

```diff
@@ -26,18 +27,40 @@ public void handleInsert(String databaseName, Document query) {
                 .withTimestamp(new BsonTimestamp(instant.toEpochMilli()))
                 .withWall(instant)
                 .withOperationType(OperationType.INSERT)
-                .withOperationDocument(d)
+                .withOperationDocument(d.clone())
                 .withNamespace(String.format("%s.%s", databaseName, query.get("insert")))
                 .asDocument()).collect(Collectors.toList());
         collection.insertDocuments(oplogDocuments);
     }
 
     @Override
-    public void handleUpdate(String databaseName, Document query) {
+    public void handleUpdate(String databaseName, Document query, List<Object> updatedIds) {
+        Instant instant = clock.instant();
+        Document updateDoc = ((List<Document>) query.get("updates")).get(0);
```

Comment:
there's always only one? we should assert that with a Precondition.

Reply:
That is the kind of things that is hard to find out, as it is internals of how mongo works there is no clear documentation. I tried a few queries, updateOne, updateMany, replaceOne and it was only one for every case.
I'm yet to encounter a query that breaks it, all the tests pass.